### PR TITLE
AWS Comprehend: Added sanitize options; added default settings.

### LINKED ISSUE #260: Do the TODOs
**Is your enhancement related to a problem? Please describe.**
Current count of seven different [TODOs in the code](https://github.com/10up/classifai/search?q=TODO&type=code).  Let's look to work on resolving and DOing those.

**Describe the solution you'd like**
- [x] `Classifai/Providers/Azure/ComputerVision.php`: Implement reset_settings() method.
- [x] `Classifai/Providers/AWS/Comprehend.php`: Implement reset_settings() method.
- [ ] `Classifai/Providers/AWS/Comprehend.php`: Implement can_register() method.
- [x] `Classifai/Providers/AWS/Comprehend.php`: Implement sanitize_settings() method.
- [ ] `Classifai/Command/ClassifaiCommand.php`: gc
- [x] `Classifai/Providers/Provider.php`: Implement sanitize_settings() method.
- [x] `Classifai/Providers/Watson/NLU.php`: Implement can_register() method.
- [x] `Classifai/Providers/Watson/NLU.php`: Is this function, `check_license_key`, supposed to be here?
- [ ] `Classifai/Services/ServicesManager.php`: remove this requirement (see: `can_register`).

**Designs**
Probably none, but won't know until we dig into the TODOs.

**Describe alternatives you've considered**
Don't do the TODOs.  In which case, are they TODONTs?

**Additional context**
n/a

### PR DESCRIPTION
**Pull request by jamesmorrison**
### Description of the Change

- Added sanitize options method
- Added default settings method
 
### Alternate Designs

N/A

### Benefits

Settings are sanitized on save.
Default settings are available.

### Possible Drawbacks

N/A

### Verification Process

No code errors.
Plugin continues to function correctly.

### Checklist:

- [x] I have read the [**CONTRIBUTING**](/CONTRIBUTING.md) document.
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my change.
- [ ] All new and existing tests passed.

### Applicable Issues

https://github.com/10up/classifai/issues/260

### Changelog Entry

Added: sanitize settings for AWS Comprehend.
Added: default settings for AWS Comprehend.

### CODE REVIEW

File: includes/Classifai/Providers/Watson/NLU.php [hunk 1]
```diff
@@ -104,8 +104,16 @@ public function reset_settings() {
...
 	public function can_register() {
-		// TODO: Implement can_register() method.
-		return true;
+		$options = get_option( $this->get_option_name() );
+
+		$can_register = true;
+
+		// False where required options do not have a value saved.
+		if ( empty( $options ) || empty( $options['credentials'] ) || empty( $options['credentials']['watson_url'] ) || empty( $options['credentials']['watson_password'] ) ) {
+			$can_register = false;
```
Comment (dinhtungdu) [inline includes/Classifai/Providers/Watson/NLU.php:114]: This prevents script loading on the Watson setting page, which is used to toggle authentication methods.

**Review by dinhtungdu — CHANGES_REQUESTED** [review]
@jamesmorrison Thanks so much for the PR! The code looks good to me. There is a minor issue on the Language Processing setting page, can you please take a look?

File: includes/Classifai/Providers/Watson/NLU.php (hunk 1 above)
Comment (rahulsprajapati) [reply includes/Classifai/Providers/Watson/NLU.php:114]: Looks like this is resolve in: https://github.com/10up/classifai/pull/313/commits/7de00c1b492e652760bc855f83e73a2974246258

ref PR: https://github.com/10up/classifai/pull/313

**Comment by rahulsprajapati** [comment]
@jeffpaul, @dinhtungdu , @jamesmorrison 
 
Looks like most of the changes from this PR is merged or conflicted with recent discussion on https://github.com/10up/classifai/issues/260
I've cherry-picked remaining changes from this PR to https://github.com/10up/classifai/pull/317 so we can close https://github.com/10up/classifai/issues/260 issue. 

- Added sanitize options; added default settings: https://github.com/10up/classifai/pull/308/commits/2a4611ad2a222cbda8bfa5232e3169f2d9b6e0ed
   - As per last discussion we are removing aws provider as it's not being used.
   - ref: Issue Discussion:https://github.com/10up/classifai/issues/260#issuecomment-988863880 PR:https://github.com/10up/classifai/pull/317
-  NLU: Added can_register method checks against required options: https://github.com/10up/classifai/pull/308/commits/7474d2e706b6b124ac67f8c806acd7e1f46d1366
    - Looks like this change is already merged with PR: https://github.com/10up/classifai/pull/313, commit ref: https://github.com/10up/classifai/commit/7de00c1b492e652760bc855f83e73a2974246258
- Switch parent class sanitize_settings to abstract function: https://github.com/10up/classifai/pull/308/commits/5a7e8e08c411746664d1f68eecbeb121e751370e
     - We have this one change we will need, I've cherry-picked the commit to PR: https://github.com/10up/classifai/pull/317, commit: https://github.com/10up/classifai/pull/317/commits/3b6471fd684872fc3e1dff32f9ad26664db66010 to close https://github.com/10up/classifai/issues/260

**Comment by jeffpaul** [comment]
Closing via #313 & #317, will ensure 317 includes credit for folks who worked on this PR.

**Pull request closed without merging**